internal/topology/check/compatibility.go

```diff
@@ -278,19 +280,19 @@ func MachineDeploymentTopologiesAreUniqueAndDefinedInClusterClass(desired *clust
 		if !machineDeploymentClasses.Has(md.Class) {
 			allErrs = append(allErrs,
 				field.Invalid(
-					field.NewPath("spec", "topology", "workers", "machineDeployments").Index(i),
-					md,
-					fmt.Sprintf("MachineDeployment clusterClass must be defined in the ClusterClass. MachineDeployment with clusterClass %q not found in ClusterClass %v.",
-						md.Name, clusterClass.Name),
+					field.NewPath("spec", "topology", "workers", "machineDeployments").Index(i).Child("class"),
+					md.Class,
+					fmt.Sprintf("MachineDeploymentClass with name %q does not exists in ClusterClass %q",
+						md.Class, clusterClass.Name),
 				),
 			)
 		}
 		if names.Has(md.Name) {
 			allErrs = append(allErrs,
 				field.Invalid(
-					field.NewPath("spec", "topology", "workers", "machineDeployments").Index(i),
-					md,
-					fmt.Sprintf("MachineDeploymentTopology names should be unique. MachineDeploymentTopology with name %q is defined more than once.", md.Name),
+					field.NewPath("spec", "topology", "workers", "machineDeployments").Index(i).Child("name"),
+					md.Name,
+					fmt.Sprintf("MachineDeployment name should be unique. MachineDeployment with name %q is defined more than once", md.Name),
```

Comment:
I think the phrase `MachineDeployment name` is misleading as it's the name of the `MachineDeploymentTopology` and the actual MachineDeployment name will be different

Reply:
The user is changing a yaml with a field named `spec.topology.workers.machineDeployments` that contains a list of items, it gets an error on one of the items, and IMO this item should be called `machineDeplyment` to match the user PoV.

If we call this item `MachineDeploymentTopology` we are surfacing an implementation detail, the name of the type, and this can be confusing for the user looking at the YAML only. This will raise immediately the question of why the list is called `machineDeployments` and not `machineDeploymentTopologies`

Reply:
Yup you're right, too :). Sounds like there is no ideal solution

Reply:
An alternative could be to just change it to:
```go
fmt.Sprintf("name should be unique. Name %q is defined more than once", md.Name),
```
It's also less verbose and the context should be very clear as the full fieldPath is prefixed to the error.


But up to you, just a suggestion. We can also keep the current version.

Reply:
fixed

Reply:
fixed